src/main/java/io/fabric8/maven/docker/util/ContainerNamingUtil.java

```diff
@@ -57,33 +57,37 @@ public static String formatContainerName(final ImageConfiguration image,
     }
 
     /**
-     * Keep only the entry with the higest index if an indexed naming scheme for container has been chosen.
-     * @param image the image from which to the the container pattern
-     * @param buildTimestamp the timestamp for the build
-     * @param containers the list of existing containers
-     * @return a list with potentially lower indexed entries removed
+     * Filters a given list of containers so that the they can be stopped by the stop mojo (analog
+     * {@link #formatContainerName(ImageConfiguration, String, Date, Collection)} but with stopping them in mind).<br>
+     * <br>
+     * The placeholders of the containerNamePattern are resolved as follows:<br>
+     * <ul>
+     *     <li>%a is replaced with the image alias</li>
+     *     <li>%n is replaced with the image name</li>
+     *     <li>%t is replaced with any date (regex \d{10,})</li>
+     *     <li>%i is replaced with any number (regex \d+)</li>
+     * </ul>
+     * @param image given image configuration
+     * @param containers all existing/running container instances of a particular image
+     * @return filtered container instances, maybe empty but never null
      */
     public static Collection<Container> getContainersToStop(final ImageConfiguration image,
                                                             final String defaultContainerNamePattern,
-                                                            final Date buildTimestamp,
                                                             final Collection<Container> containers) {
 
         String containerNamePattern = extractContainerNamePattern(image, defaultContainerNamePattern);
 
-        // Only special treatment for indexed container names
-        if (!containerNamePattern.contains(INDEX_PLACEHOLDER)) {
-            return containers;
-        }
-
-        final String partiallyApplied =
-            replacePlaceholders(
-                containerNamePattern,
-                image.getName(),
-                image.getAlias(),
-                buildTimestamp);
+        Map<String, FormatParameterReplacer.Lookup> lookups = new HashMap<>();
+        lookups.put("a", () -> image.getAlias());
+        lookups.put("n", () -> cleanImageName(image.getName()));
+        lookups.put("t", () -> "\\d{10,}");
+        lookups.put("i", () -> "\\d+");
 
-        return keepOnlyLastIndexedContainer(containers, partiallyApplied);
+        String appliedContainerNamePattern = new FormatParameterReplacer(lookups).replace(containerNamePattern);
```

Comment:
Is it correct that this changes the current behaviour in a non-backwards compatible way? (i.e. that all containers are stopped now by default, not the latest started)

If so we would need to be more careful to keep the old behaviour (at least for a certain deprecation period) and introduce this new behaviour either by a certain indication in the container name pattern or if this is not possible, with a new flag for `docker:stop`.